src/Service.Tests/Unittests/RuntimeConfigLoaderJsonDeserializerTests.cs

```diff
@@ -379,6 +486,143 @@ public static string GetModifiedJsonString(string[] reps, string enumString)
 ";
         }
 
+        /// <summary>
+        /// Config containing variable data-source.
+        /// </summary>
+        private static string _configWithVariableDataSource = @"
+            {
+              ""$schema"": ""/dab-schema.json"",
+              ""data-source"": {0},
+              ""runtime"": {
+                ""rest"": {
+                  ""enabled"": true,
+                  ""path"": ""/api""
+                },
+                ""graphql"": {
+                  ""allow-introspection"": true,
+                  ""enabled"": true,
+                  ""path"": ""/graphql""
+                },
+                ""host"": {
+                  ""mode"": ""development"",
+                  ""cors"": {
+                    ""origins"": [],
+                    ""allow-credentials"": false
+                  },
+                  ""authentication"": {
+                    ""provider"": ""AzureAD""
+                  }
+                }
+              },
+              ""entities"": {}
+            }
+            ";
+
+        /// <summary>
+        /// Returns a datasource option for a given database type.
+        /// MSSQL options has a boolean value, CosmosDB options has string value,
+        /// MySQL datasource option is empty, PostgreSQL datasource option is null,
+        /// and DWSQL has no options.
+        /// </summary>
+        private static string GetDataSourceConfigForGivenDatabase(string databaseType)
+        {
+            string options = "";
+            string databaseTypeEnvVariable = "";
+
+            switch (databaseType)
+            {
+                case "mssql":
+                    databaseTypeEnvVariable = $"@env('MSSQL_DB_TYPE')";
+                    options = @",""options"": { ""set-session-context"": ""@env('MSSQL_SET_SESSION_CONTEXT')"" }";
+                    break;
+                case "cosmosdb_nosql":
+                    databaseTypeEnvVariable = $"@env('COSMOS_DB_TYPE')";
+                    options = @",
+                        ""options"": { 
+                            ""database"": ""@env('DATABASE_NAME')"",
+                            ""container"": ""@env('DATABASE_CONTAINER')"",
+                            ""schema"": ""@env('GRAPHQL_SCHEMA_PATH')"" 
+                          }";
+                    break;
+                case "mysql":
+                    databaseTypeEnvVariable = $"@env('MYSQL_DB_TYPE')";
+                    options = @",""options"": {}";
+                    break;
+                case "postgresql":
+                    databaseTypeEnvVariable = $"@env('POSTGRESQL_DB_TYPE')";
+                    options = @",""options"": null";
+                    break;
```

Comment:
by including values for `options` when the dbtype doesn't support more options, are you testing that DAB ignores those fields?

Reply:
the role of serializer is just to correctly translate the values from the config. This test is checking that we support different ways in which options can be provided.

Reply:
if options are provided that is not required for a db, it will be ignored, but serializer role is just to translate the values correctly.